auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -694,12 +789,18 @@ private Void createPrivileged() throws RealmUnavailableException {
                 final XMLOutputFactory xmlOutputFactory = XMLOutputFactory.newFactory();
                 try (OutputStream outputStream = new BufferedOutputStream(Files.newOutputStream(tempPath, WRITE, CREATE_NEW, DSYNC))) {
                     try (AutoCloseableXMLStreamWriterHolder holder = new AutoCloseableXMLStreamWriterHolder(xmlOutputFactory.createXMLStreamWriter(outputStream))) {
+                        String namespace = "";
+                        if (secretKey != null || privateKey != null) {
```

Comment:
If `secretKey != null`, version 1.2 should be used.
Otherwise, if `privateKey != null`, version 1.1 should be used.
In all other cases, version 1.0 should be used.

Reply:
Did you swap the `secretKey` and `privateKey`? I assume if there is a private key we should use 1.2, and if theres **only** a secret key we should use 1.1

Reply:
Yes, sorry, just noticed I did swap them.